component/otelcol/receiver/vcenter/vcenter.go

```diff
@@ -0,0 +1,345 @@
+		"vcenter.datastore.disk.usage":        args.VcenterDatastoreDiskUsage.Convert(),
+		"vcenter.datastore.disk.utilization":  args.VcenterDatastoreDiskUtilization.Convert(),
+		"vcenter.host.cpu.usage":              args.VcenterHostCPUUsage.Convert(),
+		"vcenter.host.cpu.utilization":        args.VcenterHostCPUUtilization.Convert(),
+		"vcenter.host.disk.latency.avg":       args.VcenterHostDiskLatencyAvg.Convert(),
+		"vcenter.host.disk.latency.max":       args.VcenterHostDiskLatencyMax.Convert(),
+		"vcenter.host.disk.throughput":        args.VcenterHostDiskThroughput.Convert(),
+		"vcenter.host.memory.usage":           args.VcenterHostMemoryUsage.Convert(),
+		"vcenter.host.memory.utilization":     args.VcenterHostMemoryUtilization.Convert(),
+		"vcenter.host.network.packet.count":   args.VcenterHostNetworkPacketCount.Convert(),
+		"vcenter.host.network.packet.errors":  args.VcenterHostNetworkPacketErrors.Convert(),
+		"vcenter.host.network.throughput":     args.VcenterHostNetworkThroughput.Convert(),
+		"vcenter.host.network.usage":          args.VcenterHostNetworkUsage.Convert(),
+		"vcenter.resource_pool.cpu.shares":    args.VcenterResourcePoolCPUShares.Convert(),
+		"vcenter.resource_pool.cpu.usage":     args.VcenterResourcePoolCPUUsage.Convert(),
+		"vcenter.resource_pool.memory.shares": args.VcenterResourcePoolMemoryShares.Convert(),
+		"vcenter.resource_pool.memory.usage":  args.VcenterResourcePoolMemoryUsage.Convert(),
+		"vcenter.vm.cpu.usage":                args.VcenterVMCPUUsage.Convert(),
+		"vcenter.vm.cpu.utilization":          args.VcenterVMCPUUtilization.Convert(),
+		"vcenter.vm.disk.latency.avg":         args.VcenterVMDiskLatencyAvg.Convert(),
+		"vcenter.vm.disk.latency.max":         args.VcenterVMDiskLatencyMax.Convert(),
+		"vcenter.vm.disk.throughput":          args.VcenterVMDiskThroughput.Convert(),
+		"vcenter.vm.disk.usage":               args.VcenterVMDiskUsage.Convert(),
+		"vcenter.vm.disk.utilization":         args.VcenterVMDiskUtilization.Convert(),
+		"vcenter.vm.memory.ballooned":         args.VcenterVMMemoryBallooned.Convert(),
+		"vcenter.vm.memory.swapped":           args.VcenterVMMemorySwapped.Convert(),
+		"vcenter.vm.memory.swapped_ssd":       args.VcenterVMMemorySwappedSsd.Convert(),
+		"vcenter.vm.memory.usage":             args.VcenterVMMemoryUsage.Convert(),
+		"vcenter.vm.memory.utilization":       args.VcenterVMMemoryUtilization.Convert(),
+		"vcenter.vm.network.packet.count":     args.VcenterVMNetworkPacketCount.Convert(),
+		"vcenter.vm.network.throughput":       args.VcenterVMNetworkThroughput.Convert(),
+		"vcenter.vm.network.usage":            args.VcenterVMNetworkUsage.Convert()}
+}
+
+type ResourceAttributeConfig struct {
+	Enabled bool `river:"enabled,attr"`
+}
+
+// SetToDefault implements river.Defaulter.
+func (args *ResourceAttributeConfig) SetToDefault() {
+	*args = ResourceAttributeConfig{
+		Enabled: true,
+	}
+}
```

Comment:
I think we can remove this because "enabled" is mandatory. Hence, there is no need for a default value.